src/hooks/useSpeech.js

```diff
@@ -0,0 +1,92 @@
+import { noop } from 'lodash';
+import { useEffect, useMemo, useState } from 'react';
+import useIsMounted from './useIsMounted';
+
+// Used when we bail out of the hook to provide a consistent surface for consumers.
+const noopReturnValues = {
+  speak: noop,
+  stop: noop,
+  error: null,
+  isSpeaking: false,
+};
+
+/**
+ * Hook for text-to-speech.
+ *
+ * Will eventually be replaced by Google's TTS cloud system. Just a fun experiment
+ * for now.
+ */
+const useSpeech = (text, lang = window.navigator.language) => {
+  const [isSpeaking, setIsSpeaking] = useState(false);
+  const [error, setError] = useState(null);
+  const isMounted = useIsMounted();
+
+  // No text means nothing to do.
+  if (!text) {
+    return noopReturnValues;
+  }
+
+  // Speech synthesis API isn't supported on Android: https://bugs.chromium.org/p/chromium/issues/detail?id=487255
+  if (!('speechSynthesis' in window)) {
+    return {
+      ...noopReturnValues,
+      error: 'Speech API not supported',
+    };
+  }
+
+  const voices = useMemo(() => {
+    if (error) {
+      return {};
+    }
```

Comment:
This can be removed, because we don't set the error state when speech synthesis isn't available on the platform - we just return early.